### ISSUE TITLE
NPE in DefaultSessionResponseFactory.getValidUntil - No Login possible

### ISSUE THREAD

**Issue by HenryTheSir** [Graylog2/graylog2-server#13542]
## Expected Behavior

Prompt for Login
## Current Behavior

No Cookies or Session Storage existing while request is made

![grafik](https://user-images.githubusercontent.com/11004513/192225663-d5e0807b-961f-471e-8eb0-eaf39fb1cb03.png)
Cycling forever

/api/system/sessions Return HTTP 500:
{"type":"ApiError","message":""}

Stacktrace in Backend:
```
2022-09-26 10:04:58,245 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
java.lang.NullPointerException: null
        at org.graylog2.rest.models.system.sessions.responses.DefaultSessionResponseFactory.getValidUntil(DefaultSessionResponseFactory.java:52) ~[graylog.jar:?]
        at org.graylog.plugins.security.authservice.rest.OauthSessionResponseFactory.forSession(OauthSessionResponseFactory.java:28) ~[?:?]
        at org.graylog2.rest.resources.system.SessionsResource.validateSession(SessionsResource.java:188) ~[graylog.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391) ~[graylog.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80) ~[graylog.jar:?]
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255) [graylog.jar:?]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [graylog.jar:?]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [graylog.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [graylog.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [graylog.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [graylog.jar:?]
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [graylog.jar:?]
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [graylog.jar:?]
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680) [graylog.jar:?]
        at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [graylog.jar:?]
        at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

```
## Possible Solution

Catch NPE and Return not logged in message?
## Steps to Reproduce (for bugs)

All steps are made with LDAP Backend Config
1.  Login to api (basic-auth) e.g. <glhost>/api/system/notifications 
2.  Page returns successfull
3. switch to normal frontend
4. get prompted for login
5. login cycles forever
6. on page reload page wil cycle now always forever

## Your Environment

* Graylog Version: Graylog 4.3.4 and Graylog 4.3.6
* Java Version: 11.0.x
* Elasticsearch Version: %
* MongoDB Version: % 
* Operating System: RHEL 7/8
* Browser version: Firefox ESR

**Comment by HenryTheSir** [comment]
It seams reproducable:

I updated the steps in the opener

**Comment by dennisoelkers** [comment]
Did this happen for previous versions for you? What was the last version that worked?

**Comment by HenryTheSir** [comment]
It did not occur in previous versions, but I am unsure if this constelation ever happend before (logged in via direct API Call/basic auth and then switching to normal frontend without active cookie/session storage based session ). 

The problem seams to be the Basic Auth Authorization Header, which gets wrongly interpreted as a valid session in OauthSessionResponseFactory.

**Comment by mpfz0r** [comment]
Fixed with #15298
